FernFlower-Patches/0028-Improve-inferred-generic-types.patch

```diff
@@ -466,6 +476,39 @@ index dd3cde1a35d189ad7ea847c9f5de2531e4177a1b..ae78c8ccbdb5240c2e72a2700f63e373
 +            if (mask == null || mask.get(i) == null) {
 +              VarType paramType = desc.getSignature().parameterTypes.get(j++);
 +              if (paramType.isGeneric()) {
++                Exprent parameter = parameters.get(i);
++                Set<VarType> excluded = new HashSet<>();
++                if (parameter.type == Exprent.EXPRENT_NEW) {
++                  NewExprent newExprent = (NewExprent) parameter;
++                  if (newExprent.isLambda()) {
++                    ClassNode node = DecompilerContext.getClassProcessor().getMapRootClasses().get(newExprent.getNewType().getValue());
++                    int potentialMethodCount = Integer.MAX_VALUE;
++                    if (node.lambdaInformation.is_method_reference) {
++                      StructClass content = (StructClass) DecompilerContext.getStructContext().getClass(node.lambdaInformation.content_class_name);
++                      StructClass currentCls = (StructClass) DecompilerContext.getProperty(DecompilerContext.CURRENT_CLASS);
++                      potentialMethodCount = (int) content.getMethods().stream().filter((method) -> canAccess(currentCls, method)).map(StructMethod::getName).filter(node.lambdaInformation.content_method_name::equals).count();
```

Comment:
would you break your stream bits here (and below) up onto separate lines? it's a touch easier to read that way